src/main/java/agents/leeroy/Leeroy.java

```diff
@@ -66,11 +71,109 @@ private void setPhase(Risk game) {
     }
 
     private RiskAction selectInitialCountry(Risk game) {
-        performMCTS(initialPlacementRoot, GameUtils.partialInitialExpansionFunction(game), GameUtils.partialInitialEvaluationFunction(game));
-        var bestNode = initialPlacementRoot.getSuccessors().stream().max(Comparator.comparingDouble(Node::getWinScore)).get();
         //Graph moved one node forward after the action
-        initialPlacementRoot = bestNode;
-        return RiskAction.select(bestNode.getId());
+        initialPlacementRoot = searchBestNode(initialPlacementRoot, GameUtils.partialInitialExpansionFunction(game), GameUtils.partialInitialEvaluationFunction(game));
+        return RiskAction.select(initialPlacementRoot.getId());
+    }
+
+    private RiskAction reinforce(Risk game) {
```

Comment:
would probably make sense to extract this into a dedicated class. I'd like to transitition to a plug-in approach (e.g. with behavior-patterns or smth like that) which would allow us more easily to compare different combinations against each other. Let's discuss tomorrow :)